Source.cpp

```diff
@@ -0,0 +1,94 @@
+#include<iostream>
+#include"Player.h"
+
+//DECLARATION
+void Fight(Player*, Player*);
+void PrintAllStatus(Player*, Player*);
+void CheckHp(Player*, Player*, bool &);
+
+
+
+int main() 
+{
+	//Creating players 	
+	Player* player1 = new Player("Maple",150,10);
+	Player* player2 = new Player("Sally",45,30);
+	
+	//Start fight 
+	Fight(player1,player2);
+
+	//Deleting pointers
+	delete player1;
+	delete player2;
+
+	return 0;
+}
+
+//DEFINITION
+void Fight(Player* player1, Player* player2)
+{
+	//fip = fight in progress
+	bool fip = true;
+	bool playerSwap = true;
+	
+	//Printng all starting status 
+	PrintAllStatus(player1, player2);
+
+	//Fight scene 
+	while (fip)
+	{
+		//1th->2th
+		if(playerSwap)
+		{
+			std::cout << player1->GetName() << " -> " << player2->GetName()<< "\n";
+
+			//Attack
+			player2->SetInjury(player1->GetDmg());
+		
+			//Status after the round 
+			PrintAllStatus(player1, player2);
+			//Check players hp
+			CheckHp(player1, player2, fip);
+			//Swap players
+			playerSwap = false;
+		}
+		//2th->1th
```

Comment:
Most komolyan, ugyanazt leírtátok kétszer, csak megcserélt sorrendben átadott paraméterekkel? Ha már úgyis mutatókkal dolgoztok, nem lett volna sokkal egyszerűbb a mutató-másolatok értékét felcserélni? Még a playerSwap változót is el lehetne hagyni...